### PR TITLE
MES-1846: GDS Checkboxes

### PR DESCRIPTION
**Pull request by johnmorleybjss**
## Description and relevant Jira numbers
* Create an example of how to do GDS checkboxes, add to the component library.
* Ensure that sizing of the elements are suitable for future margin/padding.

image

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/theme/sass-partials/_checkbox.scss [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  // Draw the checkbox itself
+  & + label:before {
+    content: '';
+    margin-right: 12px;
+    display: inline-block;
+    vertical-align: middle;
+    width: $checkbox-width-height;
+    height: $checkbox-width-height;
+    background: white;
+    border: $checkbox-border solid #222222;
```
Comment (ammarhaiderbjss) [inline src/theme/sass-partials/_checkbox.scss]: Can we declare these colours globally and reference them using `map-get()`?

File: src/theme/sass-partials/_checkbox.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    border-radius: 6px;
+    box-shadow: inset 0 3px 3px 0 rgba(165, 162, 162, 0.78);
+  }
+
+  &:checked + label:before {
+    background: #FCF8C8;
```
Comment (ammarhaiderbjss) [inline src/theme/sass-partials/_checkbox.scss]: Declare colour in `_colors.scss`

Comment (ammarhaiderbjss) [reply src/theme/sass-partials/_checkbox.scss]: Meant to say `_variables.scss` soz

Comment (johnmorleybjss) [reply src/theme/sass-partials/_checkbox.scss]: Will extract colours in the next PR.

**Pull request merged**